crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -902,7 +902,7 @@ fn handle_leading_class_with_decorators_comment<'a>(
     comment: DecoratedComment<'a>,
     class_def: &'a ast::StmtClassDef,
 ) -> CommentPlacement<'a> {
-    if comment.start() < class_def.name.start() {
+    if comment.line_position().is_own_line() && comment.start() < class_def.name.start() {
```

Comment:
End-of-line comments are already correctly associated, I'm guessing, since they're associated with the preceding rather than the following node?

Reply:
Yes. The default placement correctly associates them with the decorator (preceding) node.